### PR TITLE
[C-Api] Check feature before processing API

### PR DESCRIPTION
**Pull request by again4you**
This patch checks the machine_learning.inference feature before
processing API by calling ml_get_feature_enabled() function.

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test: [* ]Passed [ ]Failed []Skipped
2. Run test: [* ]Passed [ ]Failed []Skipped

### CODE REVIEW

**Comment by again4you** [comment]
### Submit v2
* Check feature before processing API.
  * If `unit test` is enabled or Non-Tizen API, checking process is skipped.
* Update documentation for new feature key of Tizen API.

File: meson_options.txt [hunk 1]
```diff
@@ -15,3 +15,4 @@ option('enable-symbolic-link', type: 'boolean', value: true)
 option('enable-capi', type: 'boolean', value: false)
 option('enable-python', type: 'boolean', value: true)
 option('enable-tizen', type: 'boolean', value: false)
+option('enable-unittest', type: 'boolean', value: false)
```
Comment (again4you) [inline meson_options.txt]: Use pre-defined `enable-test` instead of `enable-unittest`

File: api/capi/src/nnstreamer-capi-pipeline.c [hunk 2]
```diff
@@ -1093,6 +1117,9 @@ ml_pipeline_switch_release_handle (ml_pipeline_switch_h h)
 {
   handle_init (switch, swtc, h);
 
+  if (ML_ERROR_NONE != ml_get_feature_enabled())
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-pipeline.c]: change to macro check_feature_state()

File: api/capi/src/nnstreamer-capi-util.c [hunk 3]
```diff
@@ -777,3 +815,42 @@ ml_check_nnfw_availability (ml_nnfw_type_e nnfw, ml_nnfw_hw_e hw,
...
+int ml_get_feature_enabled (void)
+{
+  /** -1: Not checked yet, 0: Not supported, 1: Supported */
+  static int feature_enabled = -1;
+
+/**
+ * In case of Unit Test or Non-Tizen API, then feature checking is not necessary.
+ * So it always returns ML_ERROR_NONE;
+ */
+#if defined(__UNITTEST__)
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-util.c]: The option '-Denable-unittest=true' defines '__UNITTEST__' and this always makes the return result ML_ERROR_NONE.

File: api/capi/src/nnstreamer-capi-pipeline.c (hunk 2 above)
Comment (again4you) [reply api/capi/src/nnstreamer-capi-pipeline.c]: I will!

File: meson_options.txt (hunk 1 above)
Comment (again4you) [reply meson_options.txt]: Fixed

File: api/capi/src/nnstreamer-capi-util.c (hunk 3 above)
Comment (again4you) [reply api/capi/src/nnstreamer-capi-util.c]: Not necessary.

**Comment by again4you** [comment]
### Submit v3
* Fix items that @jaeyun-jung  mentioned.

File: debian/rules [hunk 4]
```diff
@@ -35,7 +35,7 @@ override_dh_auto_clean:
 
 override_dh_auto_configure:
 	mkdir -p build
-	meson --buildtype=plain --prefix=/usr --sysconfdir=/etc --libdir=lib/$(DEB_HOST_MULTIARCH) --bindir=lib/nnstreamer/bin --includedir=include -Dinstall-example=true -Denable-tensorflow=$(enable_tf) -Denable-capi=true build
+	meson --buildtype=plain --prefix=/usr --sysconfdir=/etc --libdir=lib/$(DEB_HOST_MULTIARCH) --bindir=lib/nnstreamer/bin --includedir=include -Dinstall-example=true -Denable-tensorflow=$(enable_tf) -Denable-unittest=false -Denable-capi=true -Denable-tizen=false build
```
Comment (jaeyun-jung) [inline debian/rules]: plz remove option '-Denable-unittest=false'  (it is unnecessary)

IMO, others are ok.

File: meson.build [hunk 5]
```diff
@@ -205,6 +205,9 @@ if get_option('enable-tizen')
   add_project_arguments('-D__TIZEN__=1', language: ['c', 'cpp'])
 endif
 
+if get_option('enable-test')
+  add_project_arguments('-D__UNITTEST__=1', language: ['c', 'cpp'])
```
Comment (myungjoo) [inline meson.build]: With the current logic, ```__UNITTEST__``` should be enabled ONLY WHEN the objects are built for gtest, but you are enabling it when the project has unit-test enabled.

You need to modify how gtest binaries are built, not the global build.

**Pull request merged**